packet/flatcar-linux/kubernetes/workers/workers.tf

```diff
@@ -3,35 +3,15 @@ resource "packet_device" "nodes" {
   hostname         = "${var.cluster_name}-${var.pool_name}-worker-${count.index}"
   plan             = "${var.type}"
   facilities       = ["${var.facility}"]
-  operating_system = "custom_ipxe"
+  operating_system = "flatcar_${var.os_channel}"
```

Comment:
Need to check as the installation process is now controlled by Packet.

Reply:
I checked, and it is not. This, basically, complicates using node local storage (if not a blocker/regression).

To check, I created some `c2.medium.x86` workers with terraform and I see this:

```
core@rata-test-pool-worker-2 ~ $ lsblk -lnpd -I 8,259 | sort -h -k 4,4
/dev/sdc   8:32   0 223.6G  0 disk 
/dev/sdd   8:48   0 223.6G  0 disk 
/dev/sda   8:0    0 447.1G  0 disk 
/dev/sdb   8:16   0 447.1G  0 disk 
core@rata-test-pool-worker-2 ~ $ cat /proc/partitions 
major minor  #blocks  name

   8       16  468851544 sdb
   8        0  468851544 sda
   8        1     131072 sda1
   8        2       2048 sda2
   8        3    1048576 sda3
   8        4    1048576 sda4
   8        6     131072 sda6
   8        7      65536 sda7
   8        9  466422599 sda9
   8       32  234431064 sdc
   8       48  234431064 sdd
 254        0    1040376 dm-0
```
So, the OS is installed in `/dev/sda` (the one that has partitions) and `/dev/sda` has 447.1 G size in this server

However, on another server created exactly in the same way, I see:

```
core@rata-test-pool-worker-4 ~ $ lsblk -lnpd -I 8,259 | sort -h -k 4,4
/dev/sda   8:0    0 223.6G  0 disk 
/dev/sdb   8:16   0 223.6G  0 disk 
/dev/sdc   8:32   0 447.1G  0 disk 
/dev/sdd   8:48   0 447.1G  0 disk 
core@rata-test-pool-worker-4 ~ $ cat /proc/partitions
major minor  #blocks  name

   8        0  234431064 sda
   8        1     131072 sda1
   8        2       2048 sda2
   8        3    1048576 sda3
   8        4    1048576 sda4
   8        6     131072 sda6
   8        7      65536 sda7
   8        9  232002119 sda9
   8       16  234431064 sdb
   8       48  468851544 sdd
   8       32  468851544 sdc
 254        0    1040376 dm-0

```
So, the OS here is installed in the 223.6G size disk on this server.

I've already reported this to packet and will have a meeting to clarify and improve this.

Reply:
I think this issue can be addressed when the process of creating Flatcar images for Packet is worked on. But I think it shouldn't block this PR.

Reply:
I'm not sure about that. This creates the following problem:

* You create two identical servers: server A and N. Let's say servers have 2 disks: disk 1 200G, disk 2 400G
* As the OS is installed to a "random" disk, let's say in one the OS is installed to the 200G disk and in the other to the 400G disk
* Your identical servers not so similar anymore: one has 400G of spare disk and the other 200G
* If you are using spare disks for Container Attached Storage, node local storage or something, you can't reliably know how much space a server will add.

This is something that is very useful for some clients already running on Packet :-/

We can either:
 * Make sure Packet improves this before merging
 * Merge this, but know that until Packet imprves this we can't use this with clients. If we are also adding features, we may have to backport-them to a branch that doesn't use this installation method (but the old one), as this ones is not ready for clients.

I lean towards not merging this until is ready for use (master should always be ready for use, IMHO), as it may complicate the workloads with clients a lot if we merge it and it is not resolved soon. And if it is resolved soon, then we can merge soon too :-D

What do others think? @johananl @invidian ?

Reply:
Packet has released new images which should fix this on all regions. I think this is solved now